### PR TITLE
Move all js library decisions to a single location

### PR DESCRIPTION
**Pull request by kripken**
This was spread out in 2 places before, needlessly.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -3156,8 +3156,6 @@ def process_libraries(libs, lib_dirs, settings_changes, input_files):
     if not found:
       libraries += shared.Building.path_to_system_js_libraries(lib)
 
-  # Certain linker flags imply some link libraries to be pulled in by default.
-  libraries += shared.Building.path_to_system_js_libraries_for_settings(settings_changes)
```
Comment (sbc100) [inline emcc.py]: I think you can also remove the settings_changes arg from this function too.

Comment (kripken) [reply emcc.py]: Thanks, done.

Comment (sbc100) [reply emcc.py]: Hmm..  you can also just remove Found completely and return on line 3150?

Comment (kripken) [reply emcc.py]: Hmm, yeah, but then we'd need to duplicate the `'SYSTEM_JS_LIBRARIES="' + ','.join(libraries) + '"'` line. Not sure which I prefer.

Comment (sbc100) [reply emcc.py]: ok lgtm

**Pull request merged**